### PR TITLE
Release 3.3.0

### PR DESCRIPTION
**Pull request by davimacedo**

### CODE REVIEW

File: ChangeLog.md [hunk 1]
```diff
@@ -1,4 +1,16 @@
 ## Changelog
+
+3.3.0
+* BREAKING: shutdown() function on client now destroy the client and future requests throw error
+* Add test cases to Client
+* Improve verbose messages
+* Add request timeout handler
```
Comment (TysonAndre) [inline ChangeLog.md:7]: Could mention something like `Add a new config setting requestTimeout (defaults to 5000 milliseconds)` to make it easier to know what changed when upgrading this

Comment (davimacedo) [reply ChangeLog.md:7]: I also included in other places that it should be mentioned.

File: doc/provider.markdown [hunk 2]
```diff
@@ -25,6 +25,8 @@ Options:
 
  - `connectionRetryLimit` {Number} The maximum number of connection failures that will be tolerated before `apn.Provider` will "give up". [See below.](#connection-retry-limit) (Defaults to: 3)
 
+- `requestTimeout` {Number} The maximum time in ms that apn will wait for a request to be responded. (Defaults to: 5000)
```
Comment (TysonAndre) [inline doc/provider.markdown:28]: Awkward grammar, consider one of these alternatives

"that apn will wait for a response to a request."
"that apn will wait for a request to be responded to."

File: index.d.ts [hunk 3]
```diff
@@ -54,6 +54,10 @@ export interface ProviderOptions {
    * The maximum number of connection failures that will be tolerated before `apn` will "terminate". (Defaults to: 3)
    */
   connectionRetryLimit?: number;
+  /**
+   * The maximum time in ms that apn will wait for a request to be responded. (Defaults to: 5000)
```
Comment (TysonAndre) [inline index.d.ts:58]: Same comment

**Review by TysonAndre — CHANGES_REQUESTED** [review]
LGTM otherwise.

I haven't noticed any issues running a slightly older version of the requestTimeout PR overnight - health checks have passed and I assume node-apn reconnected or stayed connected

File: doc/provider.markdown (hunk 2 above)
Comment (davimacedo) [reply doc/provider.markdown]: thanks. done.

File: ChangeLog.md [hunk 4]
```diff
@@ -1,4 +1,17 @@
 ## Changelog
+
+3.3.0
+* BREAKING: shutdown() function on client now destroy the client and future requests throw error
```
Comment (TomWFox) [inline ChangeLog.md:4]: If it's breaking, why not go to 4.0.0?

Comment (davimacedo) [reply ChangeLog.md:4]: it is minor, but I believe you're right.

**Comment by davimacedo** [comment]
closing in favor of https://github.com/parse-community/node-apn/pull/33

**Pull request closed without merging**